src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalPopupMenu.java

```diff
@@ -127,7 +127,7 @@ public void setActiveTerminal(String caption, String handle)
    {
       activeTerminalCaption_ = caption;
       activeTerminalHandle_ = handle;
-      toolbarButton_.setText(activeTerminalCaption_);
+      toolbarButton_.setText(trimCaption(activeTerminalCaption_));
```

Comment:
Typically we do this in CSS via `text-overflow` which allows the limit to be specified in pixels rather than characters and has some slightly better typographic treatment for the ellipsis